lib/browser/api/net.js

```diff
@@ -53,18 +53,26 @@ class IncomingMessage extends Readable {
 
   get headers () {
     const filteredHeaders = {}
-    const rawHeaders = this._responseHead.headers
-    Object.keys(rawHeaders).forEach(header => {
-      if (header in filteredHeaders && discardableDuplicateHeaders.has(header)) {
+    const { rawHeaders } = this._responseHead
+    rawHeaders.forEach(header => {
+      if (header.key in filteredHeaders && discardableDuplicateHeaders.has(header.key)) {
         // do nothing with discardable duplicate headers
       } else {
-        if (header === 'set-cookie') {
+        if (header.key === 'set-cookie') {
           // keep set-cookie as an array per Node.js rules
           // see https://nodejs.org/api/http.html#http_message_headers
-          filteredHeaders[header] = rawHeaders[header]
+          if (filteredHeaders[header.key]) {
```

Comment:
This has a bug if `header.key` is one of `Object.prototype`'s keys.

e.g.

```js
const obj = {}
const key = 'constructor'
if (obj[key]) {
  obj[key].push('foo')
}
// TypeError: obj[key].push is not a function
```

Instead, check:
```suggestion
          if (Object.prototype.hasOwnProperty(filteredHeaders, header.key)) {
```